src/monocle_apptrace/wrap_common.py

```diff
@@ -93,47 +93,61 @@ def task_wrapper(tracer: Tracer, to_wrap, wrapped, instance, args, kwargs):
 
     return return_value
 
 
-def process_span(output_processor, span, instance, args):
+def process_span(to_wrap, span, instance, args):
     # Check if the output_processor is a valid JSON (in Python, that means it's a dictionary)
-    if isinstance(output_processor, dict) and len(output_processor) > 0:
-        if 'type' in output_processor:
-            span.set_attribute("span.type", output_processor['type'])
-        else:
-            logger.warning("type of span not found or incorrect written in entity json")
-        count = 0
-        if 'attributes' in output_processor:
-            count = len(output_processor["attributes"])
-            span.set_attribute("entity.count", count)
-            span_index = 1
-            for processors in output_processor["attributes"]:
-                for processor in processors:
-                    if 'attribute' in processor and 'accessor' in processor:
-                        attribute_name = f"entity.{span_index}.{processor['attribute']}"
-                        try:
-                            result = eval(processor['accessor'])(instance, args)
-                            if result and isinstance(result, str):
-                                span.set_attribute(attribute_name, result)
-                        except Exception as e:
-                            pass
-
-                    else:
-                        logger.warning("attribute or accessor not found or incorrect written in entity json")
-                span_index += 1
-        else:
-            logger.warning("attributes not found or incorrect written in entity json")
-            span.set_attribute("span.count", count)
+    span_index = 1
+    if is_root_span(span):
+        workflow_name = get_value("workflow_name")
+        if workflow_name:
+            span.set_attribute(f"entity.{span_index}.name", workflow_name)
+        # workflow type
+        package_name = to_wrap.get('package')
+        for (package, workflow_type) in WORKFLOW_TYPE_MAP.items():
+            if (package_name is not None and package in package_name):
+                span.set_attribute(f"entity.{span_index}.type", workflow_type)
+        span_index += 1
+    if 'output_processor' in to_wrap:
+        output_processor=to_wrap['output_processor']
+        if isinstance(output_processor, dict) and len(output_processor) > 0:
+            if 'type' in output_processor:
+                span.set_attribute("span.type", output_processor['type'])
+            else:
+                logger.warning("type of span not found or incorrect written in entity json")
+            count = 0
+            if 'attributes' in output_processor:
+                count = len(output_processor["attributes"])
+                span.set_attribute("entity.count", count)
+                span_index = 1
+                for processors in output_processor["attributes"]:
+                    for processor in processors:
+                        if 'attribute' in processor:
```

Comment:
There are a lot of nested if conditions. I would say that remove these and add try except

Reply:
ravi suggested to "warning message for attribute and accessor to be seperately handled". otherwise earlier  there was one if condition.